libkmod/libkmod-signature.c

```diff
@@ -278,6 +278,9 @@ static bool fill_pkcs7(const char *mem, off_t size,
 	X509_ALGOR_get0(&o, NULL, NULL, dig_alg);
 
 	sig_info->hash_algo = pkey_hash_algo[obj_to_hash_algo(o)];
+	// hash algo has not been recognized
+	if (sig_info->hash_algo == NULL)
```

Comment:
It looks like we should rather control the accesses to pkey_hash_algo[]. We don't have holes pkey_hash_algo, so this should never return NULL - it will return garbage if obj_to_hash_algo(o) returns something outside the array range.
In fact, obj_to_hash_algo(o) returns -1 and we shouldn't just pass that to the array access.